org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -438,6 +441,13 @@ private Preferences setIncompleteClasspathSeverity(Severity severity) {
 
 	public Preferences setImportOrder(List<String> importOrder) {
 		this.importOrder = (importOrder == null || importOrder.size() == 0) ? JAVA_IMPORT_ORDER_DEFAULT : importOrder;
+		IEclipsePreferences pref = JavaLanguageServerPlugin.getEclipsePreferences();
+		StringBuffer result = new StringBuffer();
+		for (String val : importOrder) {
+			result.append(val);
+			result.append(";"); //$NON-NLS-1$
+		}
+		pref.put(CodeStyleConfiguration.ORGIMPORTS_IMPORTORDER, result.toString());
```

Comment:
pref.put(CodeStyleConfiguration.ORGIMPORTS_IMPORTORDER, String.join(";", importOrder));

Reply:
Fixed in updated version.